libs/data/bbhnet/data/dataloader.py

```diff
@@ -0,0 +1,372 @@
+            sample_rate:
+                The rate at which all relevant input data has
+                been sampled
+            batch_size:
+                Number of samples to produce during at each
+                iteration
+            batches_per_epoch:
+                The number of batches to produce before raising
+                a `StopIteration` while iterating
+            waveform_sampler:
+                An object with `.fit` and `.sample` attributes for
+                sampling interferometer responses to raw waveforms
+                for injection into background data. If left as `None`,
+                no injection will take place at data loading time and
+                `waveform_frac` must be set to 0.
+            waveform_frac:
+                The fraction of each batch that should consist
+                of injected waveforms, and be marked with a
+                `1.` in the target tensor produced during iteration.
+            min_snr:
+                Minimum SNR value for sampled waveforms. Must be
+                specified if `waveform_dataset` is not `None`.
+            max_snr:
+                Maximum SNR value for sampled waveforms. Must be
+                specified if `waveform_dataset` is not `None`.
+            glitch_sampler:
+                Object for sampling glitches for insertion (not injection)
+                into background samples at data loading time. Must
+                include a `.sample` method. If passed a string, should be
+                a path to an HDF5 file containing glitches from each
+                interferometer contained in `"H1_glitches"` and
+                `"L1_glitches"` dataset keys. Data should be
+                sampled at rate `sample_rate`. Glitch triggers
+                should sit in the middle of the time axis of
+                the array containing the glitch timeseries.
+                Glitches will be randomly sampled and inserted in
+                place of the corresponding interferometer channel
+                background at data-loading time. If left as `None`,
+                no glitches will be inserted at data loading time
+                and `glitch_frac` should be 0.
+            glitch_frac:
+                The fraction of each batch that should consist
+                of inserted glitches, marked as `0.` in the
+                target tensor produced during iteration
+            device:
+                The device on which to host all the relevant
+                torch tensors.
+        """
+
+        # sanity check our fractions
+        assert 0 <= waveform_frac <= 1
+        assert 0 <= glitch_frac <= 1
+
+        self.sample_rate = sample_rate
+        self.kernel_size = int(kernel_length * sample_rate)
+        self.batch_size = batch_size
+        self.batches_per_epoch = batches_per_epoch
+        self.device = device
+
+        # load in the background data and build time-domain

```

Comment:
I assume these background files will contain real data. How many seconds of data are we using? Are we using all of this data to calculate the asd? Are we using the same asd to whiten all samples? 

Not super relevant to code, but important questions I've had that I figured should document somewhere.